src/plugins/code-golf/index.tsx

```diff
@@ -0,0 +1,103 @@
+import { Component, jsx } from "#DCGView";
+import { Inserter, PluginController } from "../PluginController";
+import "./index.less";
+import { format } from "localization/i18n-core";
+import { IfElse } from "src/components";
+import { ExpressionModel } from "src/globals";
+
+export class ExpressionItemCostPanel extends Component<{
+  model: () => ExpressionModel;
+  el: () => HTMLDivElement;
+}> {
+  rootblock: Element | null | undefined = null;
+
+  template() {
+    const chars = () => this.props.model().latex?.length ?? 0;
+
+    return IfElse(() => chars() > 0, {
+      true: () => (
+        <div class="dsm-code-golf-char-count-container">
+          <div class="dsm-code-golf-char-count">
+            <div>
+              {() => {
+                const tempRootblock = this.props
+                  .model()
+                  .dcgView?._element._domNode?.querySelector(
+                    ".dcg-main .dcg-mq-root-block"
+                  );
+
+                if (tempRootblock) this.rootblock = tempRootblock;
+
+                if (!this.rootblock) return "0px";
+
+                if (!this.rootblock.lastChild || !this.rootblock.firstChild)
+                  return "0px";
+
+                const range = document.createRange();
+                range.setStartBefore(this.rootblock.firstChild);
+                range.setEndAfter(this.rootblock.lastChild);
+
+                const width = range.getBoundingClientRect().width;
+
+                return format("code-golf-width-in-pixels", {
+                  pixels: Math.round(width).toString(),
+                });
+              }}
+            </div>
+            <div>
+              {() => {
+                const el = this.props.model().dcgView?._element._domNode;
+
+                const tempRootblock = el?.querySelector(
+                  ".dcg-main .dcg-mq-root-block"
+                );
+                if (tempRootblock) this.rootblock = tempRootblock;
+
+                function symbolCount2(el: Element) {
```

Comment:
`symbolCount2` was just to demo the two approaches here. Doesn't need to be 2 anymore.